### ISSUE TITLE
多个Server如何能独立配置

### ISSUE THREAD

**Issue by Caixiaopig** [alexazhou/VeryNginx#106]
起因：需要每个server block独自记录access_log，所以nginx.conf中写了好几个server block，端口设为不同。
结果：verynginx的配置共享了

最好的情况是能单独配置，不会互相干扰就好了。

**Comment by alexazhou** [comment]
是说针对不同的 server 块，使用不同的 VeryNginx 配置吗？

VN 设计的时候，就是设计成全局的，然后通过不同的 matcher 来对请求进行管理。所以你可以通过设置针对域名的 matcher 来对请求进行管理，就是确实没办法分开配置就是了。

**Comment by Caixiaopig** [comment]
是的。目前看来没办法分开配置了，如果是在实际生产环境下，server 块比较多，而且matcher规则都不太一样的时候，到后面matcher及其后续的filter的配置就会比较臃肿。

**Comment by alexazhou** [comment]
目前对这种情况，确实不够方便，太臃肿了可维护性也不好

**Comment by jehuty** [comment]
我需要配置http和 https 两个server block，要求两个server block 的 location相同，域名相同。
我的location是靠URI matcher来实现的，这样需要定义两个upstream（一个对应http，另一个对应https) 和两个proxy pass（一个绑定到http的upstream，另一个绑定到https的upstream）。如果想要根据用户的请求，找到对应的proxy pass，该如何配置？
